### ISSUE TITLE
ezpublish_legacy handle Not foud error (404)

### ISSUE THREAD

**Issue by kemoc** [ezsystems/LegacyBridge#87]
in my configuration
ezplatform.yml
```
ez_publish_legacy:
    #enabled: false
    system:
        ztest:
            legacy_mode: false
```
routing.yml
```
#...
_ezplatformRepositoryFormsRoutes:
    resource: "@EzSystemsRepositoryFormsBundle/Resources/config/routing.yml"

_contentOnTheFly:
    resource: "@ContentOnTheFlyBundle/Resources/config/routing.yml"
    prefix:   '%ezpublish_rest.path_prefix%'

_ezpublishLegacyRoutes:
    resource: "@EzPublishLegacyBundle/Resources/config/routing.yml"

#<end of file>
```
My eZPlatform is v1.8.1 but this problem is also on v1.3.x

**Comment by emodric** [comment]
@kemoc Can you please describe the issue you're having in more detail? It's not really clear what the problem is.

**Comment by kemoc** [comment]
Hi @emodric I did bridge Readme.md and i disabled legacy for some siteaccess and i am surprised ezpublish 4 shows error 404 but should do it Symfony stack. For enabled legacy for other siteaccess in YML config legacy shows 404 too. I got whole layout from legacy (pagelayout + content of error module/view)

**Comment by kemoc** [comment]
Hi @emodric I did bridge Readme.md and i disabled legacy for some siteaccess and i am surprised ezpublish 4 shows error 404 but should do it Symfony stack. For enabled legacy for other siteaccess in YML config legacy shows 404 too. I got whole layout from legacy (pagelayout + content of error module/view)

**Comment by emodric** [comment]
This is due to the fact that legacy bridge does not convert legacy 404 errors into Symfony exceptions, only 401 and 403:

https://github.com/ezsystems/LegacyBridge/blob/master/bundle/LegacyResponse/LegacyResponseManager.php#L87-L100

I'm not sure why is that, maybe @lolautruche can shed some light?

**Comment by lolautruche** [comment]
There is no conversion because there is no need, as `$response->setStatusCode()` is called with the error code provided by legacy kernel.

**Comment by emodric** [comment]
@lolautruche Well, it means that you can't use Symfony 404 error templates since Symfony's `NotFoundHttpException` is never thrown and eZ Platform falls back to legacy error templates which is not ideal.

Basically it depends on the usecase. On frontend, you will want Symfony's `NotFoundHttpException` while for example on legacy admin interface with `legacy_mode` set to `false`, you need legacy to handle the 404 error.

We overode this behaviour to throw `NotFoundHttpException` as needed long time ago and never had any issues.

**Comment by lolautruche** [comment]
Correct, I guess this is perfectly valid to throw a `NotFoundHttpException` in the case of a `404`

**Comment by kemoc** [comment]
@emodric What do i do now? to wait? Could you give feedback when it will be done?

**Comment by emodric** [comment]
Well, you can try and create a PR :) Otherwise, someone else needs to dedicate some time to implement this is backwards compatible way (meaning retaining the current behaviour, and enabling 404 conversion only if specified so, for selected siteacccesses.). If you will not create a PR, I will try to find some time soon, but can't make any promises.

**Comment by emodric** [comment]
Fixed in #91
